lib/rules/no-unused-prop-types.js

```diff
@@ -14,7 +14,7 @@ var variable = require('../util/variable');
 // Constants
 // ------------------------------------------------------------------------------
 
-var DIRECT_PROPS_REGEX = /^props\s*(\.|\[)/;
+var DIRECT_PROPS_REGEX = /^(props|nextProps)\s*(\.|\[)/;
```

Comment:
I think it might be good to only allow `nextProps` to work on the specific lifecycle methods it is provided by React. `shouldComponentUpdate`, `componentWillReceiveProps`, `componentWillUpdate`. And, it would probably be better to use positional argument detection for these, since people can name the arguments anything they like.

Also, it would be good to support `prevProps` on `componentDidUpdate` in the same way.